lightningd/pay.c

```diff
@@ -751,6 +790,25 @@ static void json_pay(struct command *cmd,
 		return;
 	}
 
+	if (maxfeetok
+	    && !json_tok_double(buffer, maxfeetok, &maxfeepercent)) {
+		command_fail(cmd, "'%.*s' is not a valid double",
+			     (int)(maxfeetok->end - maxfeetok->start),
+			     buffer + maxfeetok->start);
+		return;
+	}
+	if (!(0.0 < maxfeepercent)) {
```

Comment:
Specifying 0 as maxfeepercent allows you to tell it only to use direct channels, so I think that should be allowed.

And if there's a reason for the test inversion (rounding errors?) please add a comment for those of us who are less wise!

Reply:
Not really, I am simply more used to thinking of ranges in the form "min < val < max", and the inversion of the test is simply "not in range".